src/Ocelot/Configuration/Creator/RoutesCreator.cs

```diff
@@ -21,8 +21,9 @@ public class RoutesCreator : IRoutesCreator
         private readonly IRouteKeyCreator _routeKeyCreator;
         private readonly ISecurityOptionsCreator _securityOptionsCreator;
         private readonly IVersionCreator _versionCreator;
-        private readonly IVersionPolicyCreator _versionPolicyCreator;
-        private readonly IMetadataCreator _metadataCreator;
+        private readonly IVersionPolicyCreator _versionPolicyCreator;
+        private readonly IMetadataCreator _metadataCreator;
+        private readonly ITimeoutCreator _timeoutCreator;

```

Comment:
We can consider new design approach
- don't extend params of the constructor, not injecting micro services 
- extend `IRoutesCreator` by new method
- implement new method in this class

Instead of creation of tons of micro-interfaces we can implement creator-logic in this class.
💥 Bingo!